app/database/itemStore.go

```diff
@@ -28,13 +28,17 @@ func (s *ItemStore) tableName() string {
 
 // Visible returns a view of the visible items for the given user
 func (s *ItemStore) Visible(user AuthUser) DB {
+	return s.KeepItemsVisibleBy(user, s.All())
+}
+
+// KeepItemsVisibleBy returns a subview of the visible items for the given user basing on the given view
+func (s *ItemStore) KeepItemsVisibleBy(user AuthUser, db DB) DB {
```

Comment:
I'm not a big fan of these Keep* functions. Function chaining is so much readable for such a use. We should find a solution to support these.

Reply:
fixed